message/event/uploadEvent.go

```diff
@@ -0,0 +1,42 @@
+package event
+
+import (
+	"encoding/json"
+	"net/url"
+	"strings"
+)
+
+type UploadEvent struct {
+	Time  int64  `json:"time"`
+	S3URL string `json:"s3URL"`
+}
+
+func (event *UploadEvent) ConvertToEventDetails() (*EventDetails, error) {
```

Comment:
any reason for creating a struct then calling ConvertToEventDetails and returning another struct?

could just have `Create(s3URL string, t time.Time) (*EventDetails, error)` and avoid a struct